Lib/test/libregrtest/main.py

```diff
@@ -71,11 +71,11 @@ def __init__(self, ns: Namespace):
 
         # Select tests
         if ns.match_tests:
-            self.match_tests: FilterTuple = tuple(ns.match_tests)
+            self.match_tests: FilterTuple | None = tuple(ns.match_tests)
```

Comment:
Oh, my hope was that mypy would be smart enough to visit the two code paths and infer that assignation to None gives the None type.

Reply:
Mypy used to have that behaviour, but people found that this kind of implicit inference ended up just creating a lot of confusion more often than not

Reply:
Ok, good to know.